devito/operator/operator.py

```diff
@@ -148,46 +157,32 @@ def _build(cls, expressions, **kwargs):
         if any(not isinstance(i, Eq) for i in expressions):
             raise InvalidOperator("Only `devito.Eq` expressions are allowed.")
 
-        name = kwargs.get("name", "Kernel")
-        dse = kwargs.get("dse", configuration['dse'])
-
         # Python-level (i.e., compile time) and C-level (i.e., run time) performance
         profiler = create_profile('timers')
 
-        # Lower input expressions to internal expressions (e.g., attaching metadata)
+        # Lower input expressions
         expressions = cls._lower_exprs(expressions, **kwargs)
 
-        # Group expressions based on their iteration space and data dependences
-        # Several optimizations are applied (fusion, lifting, flop reduction via DSE, ...)
-        clusters = clusterize(expressions, dse_mode=set_dse_mode(dse))
+        # Group expressions based on iteration spaces and data dependences
+        clusters = cls._lower_clusters(expressions, **kwargs)
 
-        # Lower Clusters to a Schedule tree
-        stree = st_build(clusters)
+        # Turn Clusters into a Schedule tree
+        stree = cls._lower_stree(clusters, **kwargs)
 
-        # Lower Schedule tree to an Iteration/Expression tree (IET)
-        iet = iet_build(stree)
-
-        # Instrument the IET for C-level profiling
-        iet = profiler.instrument(iet)
-
-        # Wrap the IET with a Callable
-        parameters = derive_parameters(iet, True)
-        op = Callable(name, iet, 'int', parameters, ())
-
-        # Lower IET to a Target-specific IET
-        op, target_state = cls._specialize_iet(op, **kwargs)
+        # Lower Schedule tree to an Iteration/Expression tree
+        iet, byproduct = cls._lower_iet(stree, profiler, **kwargs)
```

Comment:
Isn;it better to have something like `lower_to_iet`
Seems like the difference between `lower sth` to `lower_to_sth` ?

Reply:
What you say makes sense.
However:
WHat I see is a `y=f(x)` where `x=stree` and `y=iet` and `f=lower_iet`
So it may make absolute sense, however I would like to eliminate a bit of the ambiguity.
I think adding a line or two of docstrings with some more info along what you wrote would be better.